sequence_processing_pipeline/Pipeline.py

```diff
@@ -338,15 +344,28 @@ def get_project_info(self):
                                   self.mapping_file.groupby('project_name')}
 
             for project in sample_project_map:
-                qiita_id = project.split('_')[-1]
+                matches = search(r'(.+)_(\d+)$', str(project))
+                qiita_id = matches[2]
```

Comment:
I believe this will throw an `IndexError` if `search` does not find matches as would happen on a malformed `project`

Reply:
It may be valuable to decompose the `search` operation into a discrete function that handles edge cases. It then also makes it very easy to enumerate tests of the regex, with a pattern like

```python
def test_extract_sample_project(self):
    tests = [('foo_123', ('foo', '123')), ('bar.baz_123', ('bar.baz', '123')), ('bad', None), ...etc...]
    for test, exp in tests:
        if exp is None:
            with self.assertRaises(InvalidProject):
                extract_sample_project(test)
        else:
            obs = extract_sample_project(test)
            self.assertEqual(obs, exp)
```